### PR TITLE
fix: potential NPEs found by nullness analysis [TECH-1460]

### PR DESCRIPTION
**Pull request by jbee**
Addresses issues found https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&types=BUG&id=dhis2_dhis2-core

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/resolver/CategoryOptionGroupResolver.java [hunk 1]
```diff
@@ -113,21 +114,23 @@ private String getExpression( String expression, DimensionalItemId id, List<Stri
...
 
         for ( String cogUid : cogUidList )
         {
-            CategoryOptionGroup cog = categoryOptionGroupStore
-                .getByUid( cogUid );
-
+            CategoryOptionGroup cog = categoryOptionGroupStore.getByUid( cogUid );
+            if ( cog == null )
+            {
+                continue;
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/resolver/CategoryOptionGroupResolver.java:124]: I think we should probably not silently ignore this and instead fail early, by using `loadByUid`, and throw an exception. I think we should encapsulate these null checks a bit more.

Comment (jbee) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/resolver/CategoryOptionGroupResolver.java:124]: I was also thinking about adding a `loadById(long)` that would throw an exception so we can use lookup by long and have the exception behaviour. WDYT?

Comment (larshelge) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/resolver/CategoryOptionGroupResolver.java:124]: Yes I think that sounds good.

Comment (jbee) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/resolver/CategoryOptionGroupResolver.java:124]: Yes, meant in general - there were other cases I think.

**Review by netroms — APPROVED** [review]

**Comment by jbee** [comment]
I used `loadByUid` now where single objects were fetched. In loops I used null check so objects that are not found are skipped/ignored.

**Review by vietnguyen — APPROVED** [review]

**Pull request merged**